.github/workflows/check.yaml

```diff
@@ -0,0 +1,24 @@
+---
+name: Check 🛠
+
+on:
+  # pull_request:
+  #   types:
+  #     - opened
+  #     - synchronize
+  #     - reopened
+  #     - ready_for_review
+  #   branches:
+  #     - main
+  push:
+    branches:
+      - main
+      # - setup-workflows
+  workflow_dispatch:
+
+
+# This reusable workflow from insightsengineering/r.pkg.template will run once a PR to main branch is created.
+jobs:
```

Comment:
@walkowif : I guess this one is for exercices 4 (triggers - lines 4->12) and 5 (reusable workflow) ?
For exercice 4) we could maybe add an intermediate step playing with steps outputs variables also ? 
For example : 

```
dummy:
    name: dummy step
    id: dummy_output
    run: |
          echo "variable=toto" >> $GITHUB_OUTPUT
          
r-cmd:
    name: R CMD Check 🧬
    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
    if: ${{ steps.dummy.outputs.dummy_output }} == 'toto'
```

(but we could find maybe better example 😀 ) - I don't know maybe a dummy step checking if we have well a DESCRIPTION file ..?

Reply:
Yeah, I like the idea, let me add that.

Reply:
Okay, added.